Core/GDCore/Project/Object.cpp

```diff
@@ -20,11 +20,20 @@ namespace gd {
 
 Object::~Object() {}
 
-Object::Object(const gd::String& name_, const gd::String& type_, std::unique_ptr<gd::ObjectConfiguration> configuration_)
+Object::Object(const gd::String& name_,
+               const gd::String& type_,
+               std::unique_ptr<gd::ObjectConfiguration> configuration_)
     : name(name_), configuration(std::move(configuration_)) {
       SetType(type_);
     }
 
+Object::Object(const gd::String& name_,
+               const gd::String& type_,
+               gd::ObjectConfiguration& configuration_)
+    : name(name_), configuration(&configuration_) {
+      SetType(type_);
+    }

```

Comment:
Would it be safer to make a clone of the configuration?
It would force the asset script to destroy the given configuration thought.